### PR TITLE
lib:simulations:selection_sort:Selection sort implemented

### LINKED ISSUE #162: simulation: Selection Sort
**Selection Sort**
[https://en.wikipedia.org/wiki/Selection_sort](https://en.wikipedia.org/wiki/Selection_sort
)

**Category Proposed:** Algorithms

**Additional points/links:**
[https://www.geeksforgeeks.org/selection-sort/](https://www.geeksforgeeks.org/selection-sort/)

### PR DESCRIPTION
**Pull request by apoorvdwi**
Fixed #162

### CODE REVIEW

**Review by yashugarg — CHANGES_REQUESTED** [review]
@apoorvdwi the sorting algorithm works fine but it'd be great if you could show every iteration of the loop when searching for the next smallest element to swap.

**Review by yashugarg — APPROVED** [review]
Great work
just remove "(Bars)" form the name

File: lib/src/data/simulations.dart [hunk 1]
```diff
@@ -13,22 +14,23 @@ import 'package:simulate/src/simulations/epicycloid_curve.dart';
...
-    8: "maurer rose pattern mathematics animation",
+    1: "selection sort algorithm sorting bars ",
+    2: "bubble sort algorithm sorting bars ",
+    3: "insertion sort algorithm sorting bars ",
+    4: "rose pattern mathematics sequence ",
+    5: "fourier series mathematics ",
+    6: "lissajous curve pattern mathematics animation ",
+    7: "epicycloid curve pattern mathematics animation pencil lines ",
+    8: "epicycloid curve pattern mathematics animation ",
+    9: "maurer rose pattern mathematics animation"
```
Comment (sakshamarora1) [inline lib/src/data/simulations.dart]: Don't add the new simulation in between. Add it at the end.

File: lib/src/data/simulations.dart [hunk 2]
```diff
@@ -63,84 +65,94 @@ class Simulations with ChangeNotifier {
       ),
       SimulationCard(
         id: 1,
+        simulationName: 'Selection Sort',
```
Comment (sakshamarora1) [inline lib/src/data/simulations.dart]: Here too. Move it to the end and restore the original ids of the existing simulations.

File: lib/src/data/simulations.dart [hunk 3]
```diff
@@ -63,84 +65,94 @@ class Simulations with ChangeNotifier {
...
+        image: theme.darkTheme
+            ? 'assets/simulations/SelectionSortDark.png'
+            : 'assets/simulations/SelectionSortLight.png',
```
Comment (sakshamarora1) [inline lib/src/data/simulations.dart]: I believe we can re-use the `InsertionSort` image assets and we can remove the new ones.

**Review by sakshamarora1 — COMMENTED** [review]
Looks good! Just a few minor changes and it will be ready to be merged!

**Comment by apoorvdwi** [comment]
Applied the changes suggested by @sakshamarora1

**Review by sakshamarora1 — APPROVED** [review]

**Pull request merged**